src/ifcconvert/SvgSerializer.cpp

```diff
@@ -66,14 +68,77 @@ void SvgSerializer::write(path_object& p, const TopoDS_Wire& wire) {
 
 	bool first = true;
 	util::string_buffer path;
-	path.add("            <path style=\"stroke:black; fill:none;\" d=\"");
 	for (TopExp_Explorer edges(wire, TopAbs_EDGE); edges.More(); edges.Next()) {
 		const TopoDS_Edge& edge = TopoDS::Edge(edges.Current());
 
 		double u1, u2;
 		Handle(Geom_Curve) curve = BRep_Tool::Curve(edge, u1, u2);
 
-		const bool reversed = edge.Orientation() == TopAbs_REVERSED;
+        Handle(Standard_Type) ty = curve->DynamicType();
+        bool conical = (ty == STANDARD_TYPE(Geom_Circle) || ty == STANDARD_TYPE(Geom_Ellipse));
+        bool closed = ALMOST_THE_SAME(u1 + PI2, u2);
```

Comment:
The question is whether `u1` is always smaller than `u2`. Frankly I don't really know. I think that's why I was using `fmod()` earlier.

The docs say

> if p2 < p1 the edge will be REVERSED.

https://www.opencascade.com/doc/occt-6.9.1/refman/html/class_b_rep_builder_a_p_i___make_edge.html

But I'm not sure in what order they are returned by `BRep_Tool::Curve()` later on.

Reply:
I created some hollow circle profile defs, to see if the reversed orientation there would result in `u1 > u2`. But it didn't. Let's leave it the way it is.